src/go/k8s/pkg/resources/certmanager/certificate.go

```diff
@@ -43,14 +41,15 @@ const (
 // creating Certificate from the Issuer resource to have TLS communication supported
 type CertificateResource struct {
 	k8sclient.Client
-	scheme       *runtime.Scheme
-	pandaCluster *redpandav1alpha1.Cluster
-	key          types.NamespacedName
-	issuerRef    *cmetav1.ObjectReference
-	fqdn         string
-	commonName   CommonName
-	isCA         bool
-	logger       logr.Logger
+	scheme         *runtime.Scheme
+	pandaCluster   *redpandav1alpha1.Cluster
+	key            types.NamespacedName
+	issuerRef      *cmetav1.ObjectReference
+	fqdn           string
+	commonName     CommonName
+	isCA           bool
+	keystoreSecret resources.Resource
```

Comment:
I think only name or namespacedName should be passed in to make this more testable for the future (we don't need whole resource)

Reply:
Ok, but we don't have unit tests for certmanager package.

Reply:
sure but we should still follow some good principles :)